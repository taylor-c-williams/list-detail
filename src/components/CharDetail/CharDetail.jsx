export default function CharDetail({ character }) {
  console.log(character);
  return (
    <>
      <img src={character.photoUrl} alt={character.name} />
      <p>{character.name}</p>
      <p>Affiliation: {character.affiliation}</p>
      <p>Allies: {character.allies}</p>
      <p>Enemies: {character.enemies}</p>
    </>
  );
}
