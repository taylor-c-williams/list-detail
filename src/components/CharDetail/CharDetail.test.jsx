import { render } from '@testing-library/react';
import CharDetail from './CharDetail';

const character = {
  _id: '5cf5679a915ecad153ab68c8',
  allies: ['Royal Earthbender Guards'],
  enemies: ['Chin'],
  photoUrl:
    'https://vignette.wikia.nocookie.net/avatar/images/5/51/46th_Earth_King.png/revision/latest?cb=20130627160441',
  name: '46th Earth King',
  affiliation: ' Earth Kingdom Earth Kingdom Royal Family',
};

describe('Character Detail', () => {
  it('renders a single character & details', async () => {
    const container = render(<CharDetail character={character} />);
    expect(container).toMatchSnapshot();
  });
});
