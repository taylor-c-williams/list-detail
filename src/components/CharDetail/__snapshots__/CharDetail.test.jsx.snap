// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character Detail renders a single character & details 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section>
        <article>
          <img
            alt="46th Earth King"
            src="https://vignette.wikia.nocookie.net/avatar/images/5/51/46th_Earth_King.png/revision/latest?cb=20130627160441"
          />
          <p>
            46th Earth King
          </p>
          <p>
            Affiliation: 
             Earth Kingdom Earth Kingdom Royal Family
          </p>
          <p>
            Allies: 
            Royal Earthbender Guards
          </p>
          <p>
            Enemies: 
            Chin
          </p>
        </article>
      </section>
    </div>
  </body>,
  "container": <div>
    <section>
      <article>
        <img
          alt="46th Earth King"
          src="https://vignette.wikia.nocookie.net/avatar/images/5/51/46th_Earth_King.png/revision/latest?cb=20130627160441"
        />
        <p>
          46th Earth King
        </p>
        <p>
          Affiliation: 
           Earth Kingdom Earth Kingdom Royal Family
        </p>
        <p>
          Allies: 
          Royal Earthbender Guards
        </p>
        <p>
          Enemies: 
          Chin
        </p>
      </article>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
